app/scripts/metamask-controller.js

```diff
@@ -2417,6 +2442,17 @@ export default class MetamaskController extends EventEmitter {
 
         for (const [origin, accounts] of changedAccounts.entries()) {
           this._notifyAccountsChange(origin, accounts);
+          // When permissions are added for a given origin we need to
+          // set the network client id for that origin to the current
+          // globally selected network client id.
+          if (accounts.length > 0) {
+            this.selectedNetworkController.setNetworkClientIdForDomain(
+              origin,
+              this.networkController.state.selectedNetworkClientId,
+            );
+          } else {
+            // TODO remove domain from selectedNetworkController when permissions are removed?
```

Comment:
I don't think theres any harm in leaving this state as if the user ever reconnects to this domain it will hit the logic right above and reset the selectedNetworkClientId to the globally selected network but probably cleaner to add a method to remove a domain from state on the SelectedNetworkController. wdyt @BelfordZ

Reply:
yeah I think we are good to go without it, at least I cant think of any bug that would arise from leaving the state as it is.